src/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/GetNugetPackageBasePath.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Xml;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Tools.BootstrapTasks
+{
+	public class GetNugetPackageBasePath : Task
+	{
+		[Required]
+		public ITaskItem [] PackageConfigFiles { get; set; }
+
+		[Required]
+		public string PackageName { get; set; }
+
+		[Output]
+		public string BasePath { get; set; }
```

Comment:
The advantage is it is more conventional to use ITaskItem to represent filesystem constructs. It thus helps to emphasize *meaning*, i.e. "is this `string` some arbitrary string, or some filesystem location?" Using `ITaskItem` answers that in an unambiguous fashion, and is in keeping with Microsoft's tasks.

Now, do we have existing tasks which use `string`? Yes. However, I think that was a bug/oversight, and would be more than happy to see those replaced as well. :-)